EngineLayer/PeptideSpectralMatch.cs

```diff
@@ -158,7 +158,8 @@ public void MatchToProteinLinkedPeptides(Dictionary<CompactPeptideBase, HashSet<
             OneBasedEndResidueInProtein = Resolve(pepsWithMods.Select(b => b.OneBasedEndResidueInProtein)).Item2;
             ProteinLength = Resolve(pepsWithMods.Select(b => b.Protein.Length)).Item2;
             PeptideMonisotopicMass = Resolve(pepsWithMods.Select(b => b.MonoisotopicMass)).Item2;
-            ProteinAccesion = Resolve(pepsWithMods.Select(b => b.Protein.Accession)).Item2;
+            var pepsWithModsProteaseMatch = pepsWithMods.Where(b => b.DigestionParams.Protease == DigestionParams.Protease);
```

Comment:
sorry I think I'm wrong, try editing line 150 to
`_CompactPeptides[cpKey] = new Tuple<int, HashSet<PeptideWithSetModifications>>(_CompactPeptides[cpKey].Item1, matching[cpKey].Where(p => p.DigestionParams == this.DigestionParams));`

I think that should treat the actual cause and not just the symptom. The compact pep to PWSM matching *engine* will match the compact pep to all possible peptides from any digestion param from all files. Then for this specific PSM's matching, this line will only store the ones for this PSM where they match this PSM's digestion params, which is what you want.

I'll be glad to get rid of compact peptide...

Reply:
ok, one last thing. need to make it into a hashset.

`_CompactPeptides[cpKey] = new Tuple<int, HashSet<PeptideWithSetModifications>>(_CompactPeptides[cpKey].Item1, new HashSet<PeptideWithSetModifications>(matching[cpKey].Where(p => p.DigestionParams == this.DigestionParams)));`

Reply:
This is generating some weird crash because an index is now out of range. The results appear to be fine without this fix and reverting PSM to original